lib/CGeneralTextHandler.cpp

```diff
@@ -493,6 +493,28 @@ CGeneralTextHandler::CGeneralTextHandler()
 			logGlobal->warn("WoG file ZNPC00.TXT containing commander texts was not found");
 		}
 	}
+	
+	//sanity check: in most places we get strings using operator[] so it may lead to crash if data is not valid
+	//use highest hardcoded index to check
+	if(allTexts.size() < 741 ||
+	   arraytxt.size() < 232 ||
+	   primarySkillNames.size() < 4 ||
+	   jktexts.size() < 44 ||
+	   heroscrn.size() < 33 ||
+	   overview.size() < 16 ||
+	   colors.size() < 8 ||
+	   capColors.size() < 8 ||
+	   tcommands.size() < 33 ||
+	   hcommands.size() < 9 ||
+	   fcommands.size() < 7 ||
+	   tavernInfo.size() < 8 ||
+	   zelp.size() < 614 ||
+	   advobtxt.size() < 91 ||
+	   restypes.size() < 7 ||
+	   mines.size() < 7)
+	{
+		logGlobal->error("GeneralTextHandler: GAME DATA IS CORRUPTED, GAME MAY CRASH!");

```

Comment:
lets find a way to specify which data is corrupted and how many lines do we expect. It is very close to assert(mines.size() < 7, "Expected to have at least 7 mines");

Reply:
But assert() itself might be disabled in release builds so we need some other kind of guard. verify() for instance

Reply:
That's obvious thing, I could preform multiple checks with different messages but haven't done it intentionally. It's sanity check used for troubleshooting, it's not a tool for debugging or development.
It's a purely result of bad ideas in many places of the code (I mean operator[] for vector with hardcoded numbers without checking it's size)

Now let's think how it will be used. I believe if can help with support when people come and ask why game doesn't start or crash at random moment. Usually we ask for logs, which are useless in such type of problems. Now we have troubleshooting algorithm: see message in log? -yes-> do you use custom mods? -yes-> disable them to check if problem there. Problem disappear? -no-> you have corrupted h3data.
But I don't understand what will be use case for "expected to have 7 lines of mines". 
We have to identify and highlight problems such those gettext[356] instead of making comfortable to stay with them

Reply:
You can show generic message + some detailed one. So if you see it you can understand which file you should check

Reply:
I reproduced problem on original data - there is no sense to change something in mod archive. Actually I have doubt that mods can lead to this situation because VLC will merge files, not override them

Reply:
The actual fix will be do not crash at all.

Reply:
VLC merge only things which are described with content handlers afaik. Or does it do anything special for text files as well? I remember I fixed some translation file which had a few translations missed for instance and it caused crash. But it was some JSON

Reply:
it should merge texts, all texts are stored in few std::vector containers. With merge you can increase size of the container, but not decrease

Reply:
reverted